### PR TITLE
Added Field, Ball, Robots to SimulatedTestFixture

### LINKED ISSUE #2009: Add a way to switch between Div A and Div B fields in simulation/simulated tests
### Description of the task
as titled, we just want to be able to test things in div A

### Acceptance criteria

- [ ] as titled

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* as titled, now we can switch out field

### Testing Done
regression tests pass

### Resolved Issues
resolves #2009

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by Resocram** [comment]
Overall it looks good, I was just wondering why we had to change the simulated test fixtures? Was there something wrong before about directly calling `setBallState` and `addFriendlyRobots` etc?

**Comment by jonathanlew** [comment]
> Overall it looks good, I was just wondering why we had to change the simulated test fixtures? Was there something wrong before about directly calling `setBallState` and `addFriendlyRobots` etc?

Nothing really wrong, but I didn't want to have a consistent interface with `setField` because then it would require storing all previous robots to re-add since the simulator needs to be recreated with a new field

File: src/software/simulated_tests/simulated_test_fixture.cpp [hunk 1]
```diff
@@ -134,11 +112,20 @@ void SimulatedTestFixture::sleep(
 }
 
 void SimulatedTestFixture::runTest(
+    const Field &field, const BallState &ball,
```
Comment (itsarune) [inline src/software/simulated_tests/simulated_test_fixture.cpp:115]: Doesn't look to me that the field parameter is ever used

**Review by itsarune — CHANGES_REQUESTED** [review]
looks great except one thing

File: src/software/simulated_tests/simulated_test_fixture.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/software/simulated_tests/simulated_test_fixture.cpp:115]: fixed

**Review by Resocram — DISMISSED** [review]

**Review by itsarune — DISMISSED** [review]
👍

File: src/software/ai/hl/stp/play/ball_placement_play_test.cpp [hunk 2]
```diff
@@ -11,20 +11,22 @@
 
 class BallPlacementPlayTest : public SimulatedPlayTestFixture
 {
+   protected:
+    Field field = Field::createSSLDivisionBField();
```
Comment (akhilveeraghanta) [inline src/software/ai/hl/stp/play/ball_placement_play_test.cpp:15]: you can probably toss this in the simulated play test fixture?

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
just 1 nit from me:

File: src/software/ai/hl/stp/play/stop_play_test.cpp [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 class StopPlayTest : public SimulatedPlayTestFixture
 {
-   public:
+   protected:
```
Comment (nimazareian) [inline src/software/ai/hl/stp/play/stop_play_test.cpp:15]: I'm curious what the benefit of using `protected` here is instead of `public`?

**Review by nimazareian — COMMENTED** [review]
Looks great. I just had a clarification question.

**Review by akhilveeraghanta — DISMISSED** [review]
lgtm 💯

File: src/software/ai/hl/stp/play/ball_placement_play_test.cpp (hunk 2 above)
Comment (akhilveeraghanta) [reply src/software/ai/hl/stp/play/ball_placement_play_test.cpp:15]: resolved offline: seems to be the best way to do this

File: src/software/ai/hl/stp/play/stop_play_test.cpp (hunk 3 above)
Comment (jonathanlew) [reply src/software/ai/hl/stp/play/stop_play_test.cpp:15]: in this case, there isn't much benefit, but as a principle we should limit the access of functions as much as we can (make the interface smaller)

**Review by nimazareian — APPROVED** [review]
LGTM!

**Review by itsarune — APPROVED** [review]

**Review by Resocram — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]

**Pull request merged**